indexer/src/main.rs

```diff
@@ -139,6 +141,40 @@ async fn handle_message(
     Ok(())
 }
 
+async fn download_genesis_file(opts: &configs::Opts) -> anyhow::Result<String> {
+    let res = reqwest::get(opts.genesis_file_url()).await?;
+
+    let total_size = res.content_length().unwrap();
+
+    let file_path = format!("{}-genesis.json", opts.chain_id.to_string());
+
+    match std::fs::File::open(&file_path) {
+        Ok(_) => {
+            info!("Using existing genesis file: {}", file_path);
+        }
+        Err(_) => {
+            let mut file = std::fs::File::create(&file_path)?;
+            let mut downloaded = 0;
+            let mut stream = res.bytes_stream();
+            while let Some(chunk) = stream.next().await {
+                let chunk = chunk?;
+                downloaded = std::cmp::min(downloaded + (chunk.len() as u64), total_size);
+                info!(
```

Comment:
This was the simplest approach to showing download progress but it is very noisy. I considered using `indicatif::ProgressBar` but this doesn't integrate well with `tracing` so we wouldn't see progress in e.g. Cloud Run. I'll keep looking in to this, maybe we can log every ~10 chunks or so.

Reply:
First, we're trying to call tracing macros in the full name `tracing::info!` in the indexer code-base.

So I tried to run the code. Instead of any download progress, it just keeps the silence 

```
Mar 24 14:18:58.347  INFO indexer_for_explorer:Starting Indexer for Explorer (lake)...
Mar 24 14:18:58.348  INFO indexer_for_explorer:Starting metrics server on http://0.0.0.0:3030

```

Once I've added a proper `target: INDEXER_FOR_EXPLORER` it started to print something. So, please, ensure it works out of the box. Yeah, agree it is too noisy.

As for your concern about CloudRun, we usually don't run `from-genesis` there. 

I can come up with a few variants on how to handle it:

1. Use some fancy human-friendly progress bar there and just deal with the fact CloudRun won't like it (we can ignore it)
2. Depending on how exactly CloudRun won't like the fancy progress bar we can:
    - Add info-logs with dedicated target `info!(target: "genesis-download", ...)` and expect CloudRun to catch it since we run it using `RUST_LOG=debug`. Thus using both fancy progress bars and info-logs users are not subscribed to by default.
    - Or add an optional switch via CLI args between a human-readable progress bar and noisy naive info-logs

I don't have any strong preferences here. Though I believe everyone can survive even those noisy download logs.

Reply:
> Once I've added a proper target: INDEXER_FOR_EXPLORER it started to print something.

Woops, I've added `target` 👍🏽 

> As for your concern about CloudRun, we usually don't run from-genesis there.

This might be the case now, but with the constant re-genesis on `betanet` I'd expect that we would run the `betanet` cloud run instance `from-genesis` permanently?

> Though I believe everyone can survive even those noisy download logs.

Yeah perhaps you're right, in a local scenario `from-genesis` won't be that frequent, and we reuse the existing genesis files so it will only download at least once. Regardless, I've updated the logic so that it logs every 100KB which has reduced the noise a bit.

Happy to keep iterating on this if you think the current solution isn't desirable :)